openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -482,35 +566,10 @@ def _process_texture(self, filepath, do_maketx, staging, linearize, force):
                     ("Paths not found on disk, "
                      "skipping hardlink: %s") % (existing,)
                 )
-
-        if do_maketx and ext != ".tx":
-            # Produce .tx file in staging if source file is not .tx
-            converted = os.path.join(staging, "resources", fname + ".tx")
-
-            if linearize:
-                self.log.info("tx: converting sRGB -> linear")
-                colorconvert = "--colorconvert sRGB linear"
-            else:
-                colorconvert = ""
-
-            # Ensure folder exists
-            if not os.path.exists(os.path.dirname(converted)):
-                os.makedirs(os.path.dirname(converted))
-
-            self.log.info("Generating .tx file for %s .." % filepath)
-            maketx(
-                filepath,
-                converted,
-                # Include `source-hash` as string metadata
-                "-sattrib",
-                "sourceHash",
-                escape_space(texture_hash),
-                colorconvert,
-            )
-
-            return converted, COPY, texture_hash
-
-        return filepath, COPY, texture_hash
+        for processor in processors:
+            processed_path = processor().process(filepath)
+            self.log.info("Generating texture file for %s .." % filepath)
+        return processed_path, COPY, texture_hash
```

Comment:
This will return only the last processed_path instead of all processed files.
Also, if no processors were active then `processed_path` would never be defined.

Reply:
Ah, you are right.  Then I suppose when need to be using `processed_path` at the end after it's been processed.